### PR TITLE
Add an article creation page

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
# What are the relevant tickets?
https://github.com/mitodl/mit-open/issues/160

# Description (What does it do?)
This PR adds support for creating article content via the frontend UI.

# How can this be tested?
1. Log in as a an admin. *Article creation currently limited to admins.*
2. Ensure  `CKEDITOR_ENVIRONMENT_ID`, `CKEDITOR_SECRET_KEY`, `CKEDITOR_UPLOAD_URL` env vars are set.
3. Visit http://localhost:8063/articles/new to create a new article.

*Note:* There is no article listing page right now, but you can view details at http://localhost:8063/articles/ARTICLE_ID and can view available articles at http://localhost:8063/api/v1/articles

### CODE REVIEW

File: frontends/api/src/hooks/articles/keyFactory.ts [hunk 1]
```diff
@@ -5,8 +5,10 @@ import { createQueryKeys } from "@lukemorales/query-key-factory"
 const articles = createQueryKeys("articles", {
   detail: (id: number) => ({
     queryKey: [id],
-    meow: "meow",
```
Comment (mbertrand) [inline frontends/api/src/hooks/articles/keyFactory.ts:8]: what was `meow` for? 😆

**Review by mbertrand — COMMENTED** [review]
Looks good.  If I'm not logged in as an admin, I can still access the article creation page at http://localhost:8063/articles/new  and click the Save button, the post request fails on the backend but with no frontend indication of that.  But when trying to access existing pages (http://localhost:8063/articles/2/) I get redirected to a bad login page (http://localhost:8063/login/?next=/articles/2/) Future PR for those issues?

File: frontends/api/src/hooks/articles/keyFactory.ts (hunk 1 above)
Comment (ChristopherChudzicki) [reply frontends/api/src/hooks/articles/keyFactory.ts:8]: 😆 I vaguely remember for some reason I was trying to see if we could other data on these factories... "meow" and "woof" are my go-to foo bars 😆.

**Comment by ChristopherChudzicki** [comment]
> Looks good. If I'm not logged in as an admin, I can still access the article creation page at http://localhost:8063/articles/new and click the Save button, the post request fails on the backend but with no frontend indication of that. But when trying to access existing pages (http://localhost:8063/articles/2/) I get redirected to a bad login page (http://localhost:8063/login/?next=/articles/2/) Future PR for those issues?

Good points. I'll make issues for them. That re-direct was put in-place before we had a login, but now that we have https://mitopen.odl.mit.edu/login/ol-oidc, etc, we can make the redirects real.

**Review by mbertrand — APPROVED** [review]
👍

**Pull request merged**